Cubical/Data/Int/Properties.agda

```diff
@@ -133,3 +100,156 @@ discreteInt (negsuc n) (negsuc m) with discreteℕ n m
+addEq (suc n) = compPath (addEq n) sucPathInt
+
+subEq : ℕ → Int ≡ Int
+subEq n i = addEq n (~ i)
+
+_+_ : Int → Int → Int
+m + pos n    = transport (addEq n) m
+m + negsuc n = transport (subEq (suc n)) m
+
+_-_ : Int → Int → Int
+m - pos zero    = m
+m - pos (suc n) = m + negsuc n
+m - negsuc n    = m + pos (suc n)
+
+-- We directly get that addition by a fixed number is an equivalence
+-- without having to do any induction!
+isEquivAddInt : (m : Int) → isEquiv (λ n → n + m)
+isEquivAddInt (pos n)    = isEquivTransport (addEq n)
+isEquivAddInt (negsuc n) = isEquivTransport (subEq (suc n))
+
+-- TODO: define multiplication by composing the addition equivalence
+-- with itself.
+
+private
+  +negsuc : ℕ → (Int → Int)
+  +negsuc 0 = predInt
+  +negsuc (suc n) z = predInt (+negsuc n z)
+
+  L0 : ∀ n z → +negsuc (suc n) z ≡ +negsuc n (predInt z)
+  L0 0 z = refl
+  L0 (suc n) z = cong predInt (L0 n z)
+
+  Lnegsuc : ∀ n z → z + negsuc n ≡ +negsuc n z
+  Lnegsuc 0 z = refl
+  Lnegsuc (suc n) z = compPath (Lnegsuc n (predInt z)) (sym (L0 n z))
+
+  L' : ∀ m n → m + (negsuc (suc n)) ≡ predInt (m + negsuc n)
+  L' m n = compPath (Lnegsuc (suc n) m) (cong predInt (sym (Lnegsuc n m)))
+
+  L3- : ∀ m n → m + (sucInt (negsuc n)) ≡ sucInt (m + (negsuc n))
+  L3- m 0 = sym (sucPred m)
+  L3- m (suc n) = compPath (sym (sucPred _)) (cong sucInt (sym (L' m n)))
+  
+  Lsuc : ∀ m n → m + (sucInt n) ≡ sucInt (m + n)
+  Lsuc m (pos n)    = refl
+  Lsuc m (negsuc n) = L3- m n
+
+  Lpred : ∀ m n → m + (predInt n) ≡ predInt (m + n)
+  Lpred m (pos zero)    = refl
+  Lpred m (pos (suc n)) = sym (predSuc _)
+  Lpred m (negsuc n)    = L' m n
+
+  ind-assoc : {A : Set} (_·_ : A → A → A) (f : ℕ → A)
+        (g : A → A) (p : ∀ a b → a · (g b) ≡ g (a · b))
+        (q : ∀ {c} → f (suc c) ≡ g (f c))
+        (base : ∀ m n → m · (n · (f 0)) ≡ (m · n) · (f 0))
+        (m n : A) (o : ℕ)
+      → m · (n · (f o)) ≡ (m · n) · (f o)
+  ind-assoc _·_ f g p q base m n 0 = base m n
+  ind-assoc _·_ f g p q base m n (suc o) = 
+    m · (n · (f (suc o))) ≡⟨ cong (_·_ m) (cong (_·_ n) q) ⟩
+    m · (n · (g (f o)))   ≡⟨ cong (_·_ m) (p n (f o))⟩
+    m · (g (n · (f o)))   ≡⟨ p m (n · (f o))⟩
+    g (m · (n · (f o)))   ≡⟨ cong g (assoc m n o)⟩
+    g ((m · n) · (f o))   ≡⟨ sym (p (m · n) (f o)) ⟩
+    (m · n) · (g (f o))   ≡⟨ cong (_·_ (m · n)) (sym q)⟩
+    (m · n) · (f (suc o)) ∎
+    where
+    assoc = ind-assoc _·_ f g p q base
+
++-assoc : ∀ m n o → m + (n + o) ≡ (m + n) + o
++-assoc m n (pos o) = ind-assoc _+_ pos sucInt Lsuc refl (λ _ _ → refl) m n o
++-assoc m n (negsuc o) = ind-assoc _+_ negsuc predInt Lpred refl Lpred m n o
+
+private
+  +pos : ℕ → Int → Int
+  +pos 0 z = z
+  +pos (suc n) z = sucInt (+pos n z)
+
+  L0+ : ∀ n z → +pos (suc n) z ≡ +pos n (sucInt z)
+  L0+ zero z = refl
+  L0+ (suc n) z = cong sucInt (L0+ n z)
+
+  Lpos : ∀ n z → z + pos n ≡ +pos n z
+  Lpos zero z = refl
+  Lpos (suc n) z = cong sucInt (Lpos n z)
+
+  LHS RHS : {A : Set} {x y : A} → (x ≡ y) → A
```

Comment:
These are not used for anything so please delete them

Reply:
OK. Will remove these. 

I just found `LHS` `RHS` very useful when proving identities, e.g. to prove `complicatedIdentity` I want to know the normal form of it's left hand side, instead of entering it manually I can do `LHS complicatedIdentity` in `C-c C-n`